File: solutions/code-carl/ch03_hashtable/T2_LC0242_IsAnagram.java
/**
 * Practice template for:
 * - LeetCode 242. Valid Anagram
 *   https://leetcode.cn/problems/valid-anagram/
 */
public class T2_LC0242_IsAnagram {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        int[] check = new int[26];
        for (int i = 0; i < s.length(); i++) {
            check[s.charAt(i) - 'a']++;
            check[t.charAt(i) - 'a']--;
        }
        for (int i : check) {
            if (i != 0) return false;
        }

        return true;
    }
}

File: templates/code-carl/ch03_hashtable/t2_lc0242_is_anagram.py
"""LeetCode 242. Valid Anagram
https://leetcode.cn/problems/valid-anagram/
"""


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        pass


if __name__ == "__main__":
    solution = Solution()
    assert solution.isAnagram("anagram", "nagaram") is True
    assert solution.isAnagram("rat", "car") is False
    print("All tests passed.")
